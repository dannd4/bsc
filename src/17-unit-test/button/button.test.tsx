import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Button from "./button";

describe("Button", () => {
  it("button render with correct text", () => {
    render(<Button>Click Me!!!</Button>);
    expect(screen.getByText("Click Me!!!")).toBeInTheDocument();
  });

  it("button call onClick when clicked", async () => {
    const handleClick = vi.fn();
    render(<Button onClick={handleClick}>Click Me!!!</Button>);

    await userEvent.click(screen.getByTestId("bsc-button"));
    expect(handleClick).toBeCalledTimes(1);
  });

  it("button render with correct className", () => {
    render(<Button className="btn-primary">Click Me!!!</Button>);
    expect(screen.getByTestId("bsc-button")).toHaveClass("btn-primary");
  });

  it("snapshot matches", () => {
    const { asFragment } = render(
      <Button onClick={() => {}} className="btn-primary">
        Click Me!!!
      </Button>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
