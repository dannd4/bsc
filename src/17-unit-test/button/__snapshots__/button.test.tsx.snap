// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > snapshot matches 1`] = `
<DocumentFragment>
  <button
    class="btn-primary"
    data-testid="bsc-button"
  >
    Click Me!!!
  </button>
</DocumentFragment>
`;
